modyn/evaluator/internal/dataset/evaluation_dataset.py

```diff
@@ -0,0 +1,128 @@
+import logging
+from typing import Callable, Generator, Iterable, Optional
+
+import grpc
+from modyn.storage.internal.grpc.generated.storage_pb2 import (  # pylint: disable=no-name-in-module
+    GetDataPerWorkerRequest,
+    GetDataPerWorkerResponse,
+    GetRequest,
+    GetResponse,
+)
+from modyn.storage.internal.grpc.generated.storage_pb2_grpc import StorageStub
+from modyn.utils.utils import (
+    BYTES_PARSER_FUNC_NAME,
+    MAX_MESSAGE_SIZE,
+    deserialize_function,
+    grpc_connection_established,
+)
+from torch.utils.data import IterableDataset, get_worker_info
+from torchvision import transforms
+
+logger = logging.getLogger(__name__)
+
+
+class EvaluationDataset(IterableDataset):
```

Comment:
As discussed, we should open an issue + add a TODO to unify this with the OnlineDataset (or build common abstractions) after #256 is merged (since that PR introduces many structural changes to the OD)

Reply:
Done, see #275 and TODOs.